pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -115,7 +115,7 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
 		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
-		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
+		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(defErr) {
```

Comment:
This check should probably be `err != nil && !IsConfigurationInvalid(defErr)`.  The check on the next line should remove `err == nil` and replace it with `defaultConfig != nil || !reflect.DeepEqual(mergedConfig, defaultConfig)`.

Looking at that change locally.